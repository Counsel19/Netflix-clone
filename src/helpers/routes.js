import React from "react";
import { Navigate, Outlet } from 'react-router-dom';

export function IsUserRedirect({ user, loggedInPath, children, ...rest }){
    return(
        !user ? children : <Navigate to={loggedInPath} replace />
    )
}

export function ProtectedRoute({ children, user, authPath, ...rest }){
    return(
        user ? <Outlet {...rest}>{children}</Outlet> : <Navigate to={authPath} replace />
    )
}