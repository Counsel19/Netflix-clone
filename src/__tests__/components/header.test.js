/* eslint-disable testing-library/no-node-access */
import React from "react";
import { fireEvent, screen, render, getByText, getByTestId } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Header } from "../../components";
import { MemoryRouter } from "react-router";
import * as ROUTES from "../../constants/routes";
import logo from "../../logo.svg";

describe("<Header />", () => {
  it("renders <Header /> component with background", () => {
    const { container } = render(
      <MemoryRouter>
        <Header data-testid="background-shown" src="/images/users/1.jpg" dontShowOnSmallViewPort >
          <Header.Frame>
            <Header.Logo to={ROUTES.HOME} src={logo} data-testid="logo" />
            <Header.ButtonLink to={ROUTES.SIGN_IN}>Sign In </Header.ButtonLink>
          </Header.Frame>
        </Header>
      </MemoryRouter>
    );

    expect(screen.getByText("Sign In")).toBeTruthy();
    expect(screen.getByTestId("background-shown")).toBeTruthy();
    expect(screen.getByTestId("logo")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders <Header /> component with background but withuot a source", () => {
    const { container } = render(
      <MemoryRouter>
        <Header data-testid="background-shown" >
          <Header.Frame>
            <Header.Logo to={ROUTES.HOME} src={logo} data-testid="logo" />
            <Header.ButtonLink to={ROUTES.SIGN_IN}>Sign In </Header.ButtonLink>
          </Header.Frame>
        </Header>
      </MemoryRouter>
    );

    expect(screen.getByText("Sign In")).toBeTruthy();
    expect(screen.getByTestId("background-shown")).toBeTruthy();
    expect(screen.getByTestId("logo")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders <Header /> component without background", () => {
    const { container } = render(
      <MemoryRouter>
        <Header bg={false}>
          <Header.Frame>
            <Header.Logo to={ROUTES.HOME} src={logo} data-testid="logo" />
            <Header.ButtonLink to={ROUTES.SIGN_IN}>Sign In </Header.ButtonLink>
          </Header.Frame>
        </Header>
      </MemoryRouter>
    );

    expect(screen.queryByTestId("background-shown")).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders the <Header.Feature /> component with populated data", () => {
    const { container } = render(
      <Header.Feature>
        <Header.FeatureCallOut>Watch Joker Now</Header.FeatureCallOut>
        <Header.Text>
          You create a new user in your Firebase project by calling the
          createUserWithEmailAndPassword method or by signing in a user for the
          first time using a federated identity provider, such as Google Sign-In
          or Facebook Login.
        </Header.Text>
        <Header.PlayButton>Play</Header.PlayButton>
      </Header.Feature>
    );

    expect(screen.getByText("Watch Joker Now")).toBeTruthy();
    expect(
      screen.getByText(
        "You create a new user in your Firebase project by calling the createUserWithEmailAndPassword method or by signing in a user for the first time using a federated identity provider, such as Google Sign-In or Facebook Login."
      )
    ).toBeTruthy();
    expect(screen.getByText("Play")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders the <Header.Frame /> and children", () => {
    const { container } = render(
      <MemoryRouter>
        <Header.Frame>
          <Header.Group>
            <Header.Logo to={ROUTES.HOME} src={logo} />
            <Header.TextLink active={true} onClick={() => {}}>Series</Header.TextLink>
            <Header.TextLink active={false} onClick={() => {}}>Films</Header.TextLink>
          </Header.Group>
          <Header.Group>
            <Header.Search searchTerm="Documentaries"
              setSearchTerm={() => {}}/>
            <Header.Profile>
              <Header.Picture
                src="/images.user/i.jpg"
                data-testid="header-image"
              />
              <Header.Dropdown>
                <Header.Group>
                  <Header.Picture
                    src="/images.user/i.jpg"
                    data-testid="header-image"
                  />
                  <Header.TextLink>Counsel</Header.TextLink>
                </Header.Group>
                <Header.Group>
                  <Header.TextLink onClick={() => {}}>Sign Out</Header.TextLink>
                </Header.Group>
              </Header.Dropdown>
            </Header.Profile>
          </Header.Group>
        </Header.Frame>
      </MemoryRouter>
    );

    expect(screen.getByTestId('search-input')).toBeTruthy();
    expect(screen.getByTestId('search-input').value).toBe("Documentaries");
    fireEvent.change(screen.getByTestId('search-input'), {target: {value: "Counsel"}});  
    // expect(screen.getByTestId('search-input').value).toBe("Counsel");
    fireEvent.click(screen.getByTestId("search-icon"));
    // expect(screen.queryByTestId('search-input')).toBeFalsy();

    expect(screen.getByText("Series")).toBeTruthy();
    expect(screen.getByText("Films")).toBeTruthy();
    expect(screen.getAllByTestId("header-image")).toBeTruthy();
    expect(screen.getByText("Counsel")).toBeTruthy();
    expect(screen.getByText("Sign Out")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
