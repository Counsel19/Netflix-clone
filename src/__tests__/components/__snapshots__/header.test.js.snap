// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders <Header /> component with background 1`] = `
<div
  class="header__Background-sc-10nv7v6-0 eIqvVw"
  data-testid="background-shown"
  src="/images/users/1.jpg"
>
   
  <div
    class="header__Container-sc-10nv7v6-10 kxtMcl"
  >
     
    <a
      href="/"
    >
      <img
        class="header__Logo-sc-10nv7v6-9 ByNmX"
        data-testid="logo"
        src="logo.svg"
      />
    </a>
    <a
      class="header__ButtonLink-sc-10nv7v6-11 iEAcCu"
      href="/signin"
    >
       
      Sign In 
    </a>
  </div>
</div>
`;

exports[`<Header /> renders <Header /> component with background but withuot a source 1`] = `
<div
  class="header__Background-sc-10nv7v6-0 bQlFV"
  data-testid="background-shown"
>
   
  <div
    class="header__Container-sc-10nv7v6-10 kxtMcl"
  >
     
    <a
      href="/"
    >
      <img
        class="header__Logo-sc-10nv7v6-9 ByNmX"
        data-testid="logo"
        src="logo.svg"
      />
    </a>
    <a
      class="header__ButtonLink-sc-10nv7v6-11 iEAcCu"
      href="/signin"
    >
       
      Sign In 
    </a>
  </div>
</div>
`;

exports[`<Header /> renders <Header /> component without background 1`] = `
<div
  class="header__Container-sc-10nv7v6-10 kxtMcl"
>
   
  <a
    href="/"
  >
    <img
      class="header__Logo-sc-10nv7v6-9 ByNmX"
      data-testid="logo"
      src="logo.svg"
    />
  </a>
  <a
    class="header__ButtonLink-sc-10nv7v6-11 iEAcCu"
    href="/signin"
  >
     
    Sign In 
  </a>
</div>
`;

exports[`<Header /> renders the <Header.Feature /> component with populated data 1`] = `
<div
  class="header__Container-sc-10nv7v6-10 header__Feature-sc-10nv7v6-12 kxtMcl kVoDtI"
>
  <h2
    class="header__FeatureCallOut-sc-10nv7v6-15 hVyXBK"
  >
    Watch Joker Now
  </h2>
  <p
    class="header__Text-sc-10nv7v6-14 kGDPTn"
  >
    You create a new user in your Firebase project by calling the createUserWithEmailAndPassword method or by signing in a user for the first time using a federated identity provider, such as Google Sign-In or Facebook Login.
  </p>
  <button
    class="header__PlayButton-sc-10nv7v6-13 gUTvSk"
  >
    Play
  </button>
</div>
`;

exports[`<Header /> renders the <Header.Frame /> and children 1`] = `
<div
  class="header__Container-sc-10nv7v6-10 kxtMcl"
>
   
  <div
    class="header__Group-sc-10nv7v6-2 dBbyGb"
  >
     
    <a
      href="/"
    >
      <img
        class="header__Logo-sc-10nv7v6-9 ByNmX"
        src="logo.svg"
      />
    </a>
    <p
      class="header__Link-sc-10nv7v6-3 cOWspU"
    >
      Series
    </p>
    <p
      class="header__Link-sc-10nv7v6-3 kZvceU"
    >
      Films
    </p>
  </div>
  <div
    class="header__Group-sc-10nv7v6-2 dBbyGb"
  >
     
    <div
      class="header__Search-sc-10nv7v6-4 hcUwOr"
    >
      <button
        class="header__SearchIcon-sc-10nv7v6-5 cPEvlY"
        data-testid="search-icon"
      >
        <img
          alt="Search"
          src="/images/icons/search.png"
        />
      </button>
      <input
        class="header__SearchInput-sc-10nv7v6-6 eywgJe"
        data-testid="search-input"
        placeholder="Search films and series"
        value="Documentaries"
      />
    </div>
    <div
      class="header__Profile-sc-10nv7v6-8 dNJMSl"
    >
       
      <button
        class="header__Picture-sc-10nv7v6-1 hGyOBF"
        data-testid="header-image"
        src="/images/users//images.user/i.jpg.png"
      />
      <div
        class="header__Dropdown-sc-10nv7v6-7 jTnsaB"
      >
         
        <div
          class="header__Group-sc-10nv7v6-2 dBbyGb"
        >
           
          <button
            class="header__Picture-sc-10nv7v6-1 hGyOBF"
            data-testid="header-image"
            src="/images/users//images.user/i.jpg.png"
          />
          <p
            class="header__Link-sc-10nv7v6-3 kZvceU"
          >
            Counsel
          </p>
        </div>
        <div
          class="header__Group-sc-10nv7v6-2 dBbyGb"
        >
           
          <p
            class="header__Link-sc-10nv7v6-3 kZvceU"
          >
            Sign Out
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
